src/knative/worker_knative.cpp

```diff
@@ -1,11 +1,47 @@
 #include "KnativeEndpoint.h"
 
+#ifdef WITH_BOOST
+#include <boost/program_options.hpp>
+namespace po = boost::program_options;
+#endif
+
 #include <util/logging.h>
 #include <worker/WorkerMain.h>
 
 using namespace knative;
 
-int main() {
+int main(int argc, char **argv) {
+
+    int port = 8080;
+
+#ifdef WITH_BOOST
+    // Parses command line args and report errors without the logger because they are destined to the user
+    try {
+        po::options_description desc("Allowed options");
+        desc.add_options()
+                ("help,h", "print help message")
+                ("port,p", po::value<int>()->default_value(port), "port the worker will listen on")
+                ;
+        po::variables_map vm;
+        po::store(po::parse_command_line(argc, argv, desc), vm);
+        if (vm.count("help")) {
+            std::cout << desc << std::endl;
+            return EXIT_SUCCESS;
+        }
+
+        port = vm["port"].as<int>();
+    }
+    catch(std::exception& e) {
+        std::cerr << e.what() << std::endl;
+        return EXIT_FAILURE;
```

Comment:
Yeah I'll remove things here. although with the number of cli utilities there is in the project, I think it's useful to have good support for command line args by default. I wanted to refactor `simple_runner` too for example, but one thing at a time on this PR, otherwise it's going to remain open forever